platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -184,6 +187,54 @@ class HangingProtocolService {
     this.protocols.set(protocolId, protocol);
   }
 
+  /** Sets the given protocol object as active.  Also sets the default
+   * for the given id to the specified protocol - allows updating which
+   * protocol is active for a given name.
+   */
+  public setActiveProtocol(id: string, protocol?: Protocol): void {
+    if (!id) {
+      throw new Error(`Protocol ${protocol} added without an id`);
+    }
+    console.log('Active protocol set', id, protocol);
+    if (protocol && this.protocols.get(id) !== protocol) {
+      this.addProtocol(id, protocol);
+    }
+    if (!this.activeProtocolIds) {
+      this.activeProtocolIds = [];
+    }
+    this.activeProtocolIds.push(id);
+  }
+
+  /** Sets the active hanging protocols set, given the extension names to load.
+   * @params extensionManager is the extension manager to load from
+   * @paprams hangingProtocols is a list of extensions to use for the
+   * active set.  If this is
+   */
+  public setActiveProtocols(
+    extensionManager: Record<string, unknown>,
+    hangingProtocol?: string[] | string
+  ): void {
+    if (typeof hangingProtocol === 'string') {
+      this.setActiveProtocols(extensionManager, [hangingProtocol]);
+      return;
+    }
+    this.activeProtocolIds = null;
+    if (!hangingProtocol) {
+      console.log('No active protocols, setting all to active');
+      return;
+    }
+    this.activeProtocolIds = [];
+    hangingProtocol.forEach(extensionProtocol => {
+      const hangingProtocolModule = extensionManager.getModuleEntry(
+        extensionProtocol
+      );
+      this.setActiveProtocol(
```

Comment:
I think here we should also check if the protocolModule.protocol is actually one protocol or an array of protocols, since it can be based on what we discussed here

https://gist.github.com/sedghi/c3f277f743690a6cb1e717319e29b0b1

Reply:
Removed the extension manager setting entirely from the HP Service, so it is just based on the id